### DISCUSSION TITLE
Compatibility with Azure Cosmos DB

### DISCUSSION THREAD

**Discussion by Xaseron** [FerretDB/FerretDB#1018] [General]
Is postgresql a hard requirement? Would it be possible to use Azure Cosmos DB?
I know CosmosDB has MongoDB compatibility but you have to pay for each collection which would result in horrendous cost for our test environment.

**Comment by AlekSi** [comment]
We do plan to support several backends. In fact, we already support both PostgreSQL and [Tigris](https://www.tigrisdata.com). But we currently don't have any plans to support Azure Cosmos DB.

**Comment by ptrfarkas** [comment]
@Xaseron Can you elaborate on your intended use case? In most cases, FerretDB replaces CosmosDB, as it provides you a MongDB-compatible database, as long as you also have a PostgreSQL or Tigris-compatible backend which FerretDB can connect to. This backend can be hosted anywhere,  If you are committed to Azure, you can run PostgreSQL on Azure and use that instance as the backend, but this really applies to any PostgreSQL-compatible database as a service out there, not just Azure.

This way you would not need to worry about the per collection pricing model at all, and no workarounds would be needed.

**Comment by Xaseron** [comment]
Just from the cost perspective would be Cosmos DB cheaper than Azure Postgres flexible Server.
